### PR TITLE
unit test for Read::has_chunk

### PR DESCRIPTION
**Pull request by phritz**
progress towards #35

### CODE REVIEW

File: src/dag/read.rs [hunk 1]
```diff
@@ -54,3 +54,39 @@ pub async fn get_head(kvr: &dyn kv::Read, name: &str) -> Result<Option<String>>
...
+            kvw.commit().await.unwrap();
+            assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
+
+            // Mayve test isolation: other txs should not affect what the read tx, which is
+            // still open, sees.
+            if del_haskey {
+                let kvw2 = kv.write().await.unwrap();
+                kvw2.del("haskey".into()).await.unwrap();
+                kvw2.commit().await.unwrap();
+                assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
```
Comment (whiten) [inline src/dag/read.rs:84]: I think this should be outside of the `if del_haskey` block.

File: src/dag/read.rs [hunk 2]
```diff
@@ -54,3 +54,39 @@ pub async fn get_head(kvr: &dyn kv::Read, name: &str) -> Result<Option<String>>
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::kv::memstore::MemStore;
+    use crate::kv::Store;
+
+    #[async_std::test]
+    async fn test_has_chunk() {
+        async fn test(hash: &str, del_haskey: bool, expect_has: bool) {
```
Comment (whiten) [inline src/dag/read.rs:66]: I find the `has` part of this `del_haskey` name sort of confusing. Maybe `delete` or `delete_key`?

File: src/dag/read.rs [hunk 3]
```diff
@@ -54,3 +54,39 @@ pub async fn get_head(kvr: &dyn kv::Read, name: &str) -> Result<Option<String>>
...
+            let kvr = kv.read().await.unwrap();
+            let r = Read { kvr };
+
+            kvw.put(&Key::ChunkData("haskey").to_string(), &vec![0u8, 1])
+                .await
+                .unwrap();
+            kvw.commit().await.unwrap();
+            assert_eq!(expect_has, r.has_chunk(&hash).await.unwrap());
+
+            // Mayve test isolation: other txs should not affect what the read tx, which is
```
Comment (whiten) [inline src/dag/read.rs:78]: Typo in "Mayve" here.

File: src/dag/read.rs (hunk 2 above)
Comment (phritz) [reply src/dag/read.rs:66]: done in https://github.com/rocicorp/repc/pull/41

File: src/dag/read.rs (hunk 3 above)
Comment (phritz) [reply src/dag/read.rs:78]: done in https://github.com/rocicorp/repc/pull/41

File: src/dag/read.rs (hunk 1 above)
Comment (phritz) [reply src/dag/read.rs:84]: done in https://github.com/rocicorp/repc/pull/41

**Pull request merged**